azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/message_queue.py

```diff
@@ -0,0 +1,142 @@
+import queue
+import logging
+from threading import Event
+
+logger = logging.getLogger(__name__)
+
+
+class MessageQueue(queue.Queue):
+    def __init__(self, on_enable, on_disable):
+        super(MessageQueue, self).__init__()
+        self._enabled = False
+        self._on_enable = on_enable
+        self._on_disable = on_disable
+
+    def enable(self):
+        self._enabled = True
+        self._on_enable()
+
+    def disable(self):
+        self._enabled = False
+        self._on_disable()
+
+    @property
+    def enabled(self):
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, v):
+        if v:
+            self.enable()
+        else:
+            self.disable()
+
+
+class MessageQueueManager(object):
+    def __init__(self, transport):
+        self.c2d_message_queue = None
+        self.input_message_queues = {}
```

Comment:
No use of weak references (for now). We still could as a helpful feature to ensure we don't eat up a ton of memory due to user error, but I figured we can add that later since it's not necessary for the core function of making this work properly. Make it work first, then make it right and all that.